File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/server/StartServer.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Net.Sockets;
using System.Threading;
using log4net.Config;
using model;
using model.validator;
using persistence.dbRepositories;
using persistence.interfaces;
using services;
using networking;
using networking2;

namespace server;

public class StartServer
{
    private static int DEFAULT_PORT = 55556;
    private static String DEFAULT_IP = "127.0.0.1";

    static void Main(string[] args)
    {
        XmlConfigurator.Configure(new System.IO.FileInfo("log4net.config"));
        Console.WriteLine("Reading properties from app.config ...");
        int port = DEFAULT_PORT;
        String ip = DEFAULT_IP;
        String? portS = ConfigurationManager.AppSettings["port"];
        if (portS == null)
        {
            Console.WriteLine("Port property not set. Using default value " + DEFAULT_PORT);
        }
        else
        {
            bool result = Int32.TryParse(portS, out port);
            if (!result)
            {
                Console.WriteLine("Port property not a number. Using default value " + DEFAULT_PORT);
                port = DEFAULT_PORT;
                Console.WriteLine("The port" + port);
            }

            String? ipS = ConfigurationManager.AppSettings["ip"];

            if (ipS == null)
            {
                Console.WriteLine("Ip property not set. Using default value " + DEFAULT_IP);
            }
            Console.WriteLine("Configuration Settings for database {0}", GetConnectionStringByName("triathlonDB"));
            IDictionary<String, string> props = new SortedList<string, string>();
            props.Add("ConnectionString", GetConnectionStringByName("triathlonDB"));
            IValidator<Referee> refereeValidator = new RefereeValidator();
            IRefereeRepository refereeRepository = new RefereeDbRepository(props, refereeValidator);
            IValidator<Trial> trialValidator = new TrialValidator();
            ITrialRepository trialRepository = new TrialDbRepository(props, trialValidator);
            IValidator<Participant> participantValidator = new ParticipantValidator();
            IParticipantRepository participantRepository = new ParticipantDbRepository(props, participantValidator);
            IValidator<Result> resultValidator = new ResultValidator();
            IResultRepository resultRepository = new ResultDbRepository(props, resultValidator);
            IServices services = new ServerImplementation(refereeRepository, trialRepository, participantRepository,
                resultRepository);
            
            Console.WriteLine("Starting server on IP {0} and port {1}", ip, port);
            SerialServer server = new SerialServer(ip, port, services);
            server.Start();
            Console.WriteLine("Server started ...");
            Console.ReadLine();
        }
    }

    static String GetConnectionStringByName(string name)
    {
        string returnValue = null!;

        ConnectionStringSettings settings = ConfigurationManager.ConnectionStrings[name];

        if (settings != null)
        {
            returnValue = settings.ConnectionString;
        }

        return returnValue;
    }
}

public class SerialServer : ConcurrentServer
{
    private IServices _server;
    private ProtoWorker _worker;

    public SerialServer(string host, int port, IServices server) : base(host, port)
    {
        _server = server;
        Console.WriteLine("SerialServer...");
    }


    protected override Thread CreateWorker(TcpClient client)
    {
        _worker = new ProtoWorker(_server, client);
        return new Thread(new ThreadStart(_worker.Run));
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonServer/src/main/java/StartServer.java
import triathlon.model.Participant;
import triathlon.model.Referee;
import triathlon.model.Result;
import triathlon.model.Trial;
import triathlon.model.validator.*;
import triathlon.networking.utils.AbstractServer;
import triathlon.networking.utils.ConcurrentServer;
import triathlon.networking.utils.ServerException;
import triathlon.persistence.dbRepositories.ParticipantDBRepository;
import triathlon.persistence.dbRepositories.RefereeDBRepository;
import triathlon.persistence.dbRepositories.ResultDBRepository;
import triathlon.persistence.dbRepositories.TrialDBRepository;
import triathlon.persistence.hibernateRepositories.ParticipantHibernateRepository;
import triathlon.persistence.hibernateRepositories.RefereeHibernateRepository;
import triathlon.persistence.hibernateRepositories.ResultHibernateRepository;
import triathlon.persistence.hibernateRepositories.TrialHibernateRepository;
import triathlon.persistence.interfaces.IParticipantRepository;
import triathlon.persistence.interfaces.IRefereeRepository;
import triathlon.persistence.interfaces.IResultRepository;
import triathlon.persistence.interfaces.ITrialRepository;
import triathlon.server.ServicesImplementation;
import triathlon.services.IServices;

import java.io.IOException;
import java.util.Properties;

public class StartServer {
    private static int defaultPort = 55555;

    public static void main(String[] args) {
        Properties serverProps = new Properties();
        try {
            serverProps.load(StartServer.class.getResourceAsStream("/server.properties"));
            System.out.println("Server properties set." + serverProps.getProperty("triathlon.jdbc.url"));
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find server.properties " + e);
            return;
        }
        Validator<Referee> refereeValidator = new RefereeValidator();
        IRefereeRepository refereeRepository = new RefereeHibernateRepository(refereeValidator);
        Validator<Trial> trialValidator = new TrialValidator();
        ITrialRepository trialRepository = new TrialHibernateRepository(trialValidator);
        Validator<Participant> participantValidator = new ParticipantValidator();
        IParticipantRepository participantRepository = new ParticipantHibernateRepository(participantValidator);
        Validator<Result> resultValidator = new ResultValidator();
        IResultRepository resultRepository = new ResultHibernateRepository(resultValidator);

        refereeRepository.save(new Referee("Alex Johnson", "alex", "alex"));
        refereeRepository.save(new Referee("David Johnson", "david", "david"));
        trialRepository.save(new Trial("Swimming",
                "Participants swim a designated distance in open water.",
                refereeRepository.findByUsername("alex")));
        trialRepository.save(new Trial("Running Segment",
                "Following the cycling, competitors embark on a running course.",
                refereeRepository.findByUsername("alex")));
        participantRepository.save(new Participant("Alex Johnson", "1234"));

        IServices services = new ServicesImplementation(refereeRepository, trialRepository, participantRepository,
                resultRepository);
        int serverPort = defaultPort;
        try {
            serverPort = Integer.parseInt(serverProps.getProperty("triathlon.server.port"));
        } catch (NumberFormatException e) {
            System.err.println("Wrong Port Number " + e.getMessage());
            System.err.println("Using default port " + defaultPort);
        }
        System.out.println("Starting server on port: " + serverPort);
        AbstractServer server = new ConcurrentServer(serverPort, services);
        try {
            server.start();
        } catch (ServerException e) {
            System.err.println("Error starting the server " + e.getMessage());
        }
    }
}
